Babylon/groups/api/groups/solution/commands/update.py

```diff
@@ -1,56 +1,73 @@
-import logging
+from logging import getLogger
 from pprint import pformat
+from typing import Optional
 
-import cosmotech_api
 from click import argument
 from click import command
 from click import make_pass_decorator
 from click import option
 from cosmotech_api.api.solution_api import SolutionApi
+from cosmotech_api.exceptions import NotFoundException
+from cosmotech_api.exceptions import UnauthorizedException
 
 from ......utils.api import get_api_file
+from ......utils.decorators import allow_dry_run
 from ......utils.decorators import require_deployment_key
 from ......utils.decorators import timing_decorator
-from ......utils.decorators import allow_dry_run
 
-logger = logging.getLogger("Babylon")
+logger = getLogger("Babylon")
 
 pass_solution_api = make_pass_decorator(SolutionApi)
 
 
 @command()
+@allow_dry_run
+@timing_decorator
 @pass_solution_api
-@argument("solution_file")
-@option("-e", "--use-solution-file", "use_solution_file", is_flag=True,
-        help="Should the path be in the solution ?")
+@argument("solution_file", type=str)
```

Comment:
Should be of type Path

Reply:
solution file can't be an click.Path

```python
converted_solution_content = get_api_file(
        api_file_path=solution_file,
        use_working_dir_file=use_working_dir_file,
        logger=logger,
 )
```

use_working_dir_file allow to use file from babylon working directory witch is not necessary the current path location

Reply:
my bad ;)